x-pack/plugins/discover_enhanced/public/actions/explore_data/kibana_url.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// TODO: Replace this logic with KibanaURL once it is available.
+// https://github.com/elastic/kibana/issues/64497
+export class KibanaURL {
```

Comment:
Since on last platform<->apparach common sentiment was to prefer functions and interfaces over classes and plan is to make it a guideline, 
should this be a function?

Reply:
At the sync we discussed to prefer exporting function instead of classes on plugin contracts. Like

```js
plugins.myPlugin.createSomething()
```

instead of

```js
new plugins.myPlugin.Something()
```

Not preferring functions over classes in general.

BTW, maybe we should reconsider that as the argument for that was to prevent users to extend the exported class. Which can be achieved through *private* constructor.

```ts
class FinalClass {
  private constructor() {}
}
```

I don't see how it relates to `KibanaURL` as it is internal detail of `share` plugin and URL generator has `urlGenerator.createUrl(state)` (which is a function) method not a class `new urlGenerator.URL(state)`.

Also, `URL` is implemented as a class in Node and in browsers; this PR does not introduce `KibanaURL`, but moves it to a separate file, so I would not want to make it a function in this PR.

But in future work, I'm happy if it is converted to a function instead, I have no preference on that.